drivers/flash/spi_nor.c

```diff
@@ -540,6 +546,104 @@ static int spi_nor_wrsr(const struct device *dev,
 	return ret;
 }
 
+#if DT_HAS_COMPAT_STATUS_OKAY(mxicy_mx25r_spi)
+
+/**
+ * @brief Read the configuration register.
+ *
+ * @note The device must be externally acquired before invoking this
+ * function.
+ *
+ * @param dev Device struct
+ *
+ * @return the non-negative value of the status register, or an error code.
+ */
+#define MX25R_CMD_RDCR 0x15
+static int mx25r_rdcr(const struct device *dev)
+{
+	uint16_t reg;
+	int ret = spi_nor_cmd_read(dev, MX25R_CMD_RDCR, &reg, sizeof(reg));
+
+	if (ret == 0) {
+		ret = reg;
+	}
+
+	return ret;
+}
+
+/**
+ * @brief Write the configuration register.
+ *
+ * @note The device must be externally acquired before invoking this
+ * function.
+ *
+ * @param dev Device struct
+ * @param cr  The new value of the configuration register
+ *
+ * @return 0 on success or a negative error code.
+ */
+static int mx25r_wrcr(const struct device *dev,
+			uint16_t cr)
+{
+	/* The configuration register bytes on the Macronix MX25R devices are
+	 * written using the Write Status Register command where the configuration
+	 * register bytes are written as two extra bytes after the status register.
+	 * First read out the current status register to preserve the value.
+	 */
+	int sr = spi_nor_rdsr(dev);
+
+	if (sr < 0) {
+		return sr;
+	}
+
+	int ret = spi_nor_cmd_write(dev, SPI_NOR_CMD_WREN);
+
+	if (ret == 0) {
+		uint8_t data[] = {
+			sr,
+			cr & 0xFF,	/* Configuration register 1 */
+			cr >> 8		/* Configuration register 2 */
+		};
+
+		ret = spi_nor_access(dev, SPI_NOR_CMD_WRSR, NOR_ACCESS_WRITE, 0, data,
+			sizeof(data));
+		spi_nor_wait_until_ready(dev);
+	}
+
+	return ret;
+}
+
+static int mx25r_configure(const struct device *dev)
+{
+	/* Low-power/high perf mode is second bit in configuration register 2 */
+	enum { MX25R_LH_SWITCH_BIT = 9 };
+
+	acquire_device(dev);
+
+	/* Read current configuration register */
+	int current_cr = mx25r_rdcr(dev);
+
+	if (current_cr < 0) {
+		return current_cr;
+	}
```

Comment:
This is quite confusing. I think it would be clearer to use:
```
	ret = mx25r_rdcr(dev);
	if (ret < 0) {
		return ret;
	}

	current_cr =  ret;
```